buildbot.mariadb.org/master-libvirt/master.cfg

```diff
@@ -235,88 +187,101 @@ f_rpm_upgrade.addStep(getScript('rpm-upgrade.sh'))
-      tags=["Centos", "rpm", "install"],
-      collapseRequests=True,
-      nextBuild=nextBuild,
-      canStartBuild=canStartBuild,
-      properties={'systemdCapability': 'yes', 'needsGalera': 'yes', 'version_name': '7', 'arch': 'centos74-amd64'},
-      factory=f_rpm_install))
-
-c['builders'].append(
-    util.BuilderConfig(name="amd64-centos-7-rpm-autobake-major-upgrade",
-      workernames=["buildbot-centos7"],
-      tags=["Centos", "rpm", "upgrade"],
-      collapseRequests=True,
-      nextBuild=nextBuild,
-      canStartBuild=canStartBuild,
-      properties={'systemdCapability': 'yes', 'needsGalera': 'yes', 'version_name': 'centos7', 'arch': 'amd64', "test_type": "major", "test_mode": "server"},
-      factory=f_rpm_upgrade))
-
-c['builders'].append(
-    util.BuilderConfig(name="amd64-centos-7-rpm-autobake-minor-upgrade",
-      workernames=["buildbot-centos7"],
-      tags=["Centos", "rpm", "upgrade"],
-      collapseRequests=True,
-      nextBuild=nextBuild,
-      canStartBuild=canStartBuild,
-      properties={'systemdCapability': 'yes', 'needsGalera': 'yes', 'version_name': 'centos7', 'arch': 'amd64', "test_type": "minor", "test_mode": "server"},
-      factory=f_rpm_upgrade))
-
-c['builders'].append(
-    util.BuilderConfig(name="amd64-fedora-35-rpm-autobake-install",
-      workernames=["bb-hz-bbw5-fedora-35"],
-      tags=["Fedora", "rpm", "install"],
-      collapseRequests=True,
-      nextBuild=nextBuild,
-      canStartBuild=canStartBuild,
-      properties={'systemdCapability': 'yes', 'needsGalera': 'yes', 'version_name': '35', 'arch': 'fedora35-amd64'},
-      factory=f_rpm_install))
+# Add the workers and builds based on the configured install builders (see constants.py)
+for builder_name in builders_install:
+    # Parse builder name
+    platform, os_name, os_version, builder_type = builder_name.split('-')[:4]
+
+    assert builder_type in ['rpm', 'deb']
+    assert platform in ['amd64', 'aarch64']
+
+    os_info_name = os_name + '-' + os_version
+    if platform == 'amd64':
+        libvirt_worker_name = 'bb-hz-bbw5-' + os_name + '-' + os_version
+        connection_url = 'qemu+ssh://buildbot@100.64.100.20:65001/system'
+    elif platform == 'aarch64':
+        libvirt_worker_name = 'bb-eq-arm1-' + os_name + '-' + os_version
+        connection_url = 'qemu+ssh://buildbot@100.64.100.17/system'
+
+    image_path = '/var/libvirt/images/' + libvirt_worker_name
+
+    c['workers'].append(worker.LibVirtWorker(libvirt_worker_name,
+        config["private"]["worker_pass"]["libvirt"],
+        util.Connection(connection_url),
+        image_path, build_wait_timeout=0, max_builds=1))
+
+    if builder_type == 'deb':
+        factory_install = f_deb_install
+        factory_upgrade = f_deb_upgrade
+        build_arch = platform
+    elif builder_type == 'rpm':
+        factory_install = f_rpm_install
+        factory_upgrade = f_rpm_upgrade
+        build_arch = os_name + os_version + '-' + platform
```

Comment:
`build_arch` isn't used, if it was mean too, deb/rpm differences look large without an obvious reason.

Reply:
Good catch. It should be used for the `arch` build variable for all worker definitions (line 242, 260)